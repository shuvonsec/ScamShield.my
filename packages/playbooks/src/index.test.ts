import { describe, expect, it } from 'vitest';
import { getPlaybook, getPlaybooks } from './index';

describe('playbooks', () => {
  it('returns playbooks', () => {
    const list = getPlaybooks();
    expect(list.length).toBeGreaterThan(0);
  });

  it('generates deterministic changes', () => {
    const playbook = getPlaybook('csp-builder');
    expect(playbook).toBeTruthy();
    const changes = playbook?.apply({ mode: 'preview' }) ?? [];
    expect(changes).toMatchSnapshot();
  });
});
