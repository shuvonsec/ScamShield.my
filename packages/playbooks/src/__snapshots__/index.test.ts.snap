// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`playbooks generates deterministic changes 1`] = `[
  {
    "description": "Add secure headers to reverse proxy configuration.",
    "diff": "Content-Security-Policy: default-src 'self'; img-src 'self' data:; script-src 'self'; object-src 'none'\\nReferrer-Policy: strict-origin-when-cross-origin\\nX-Frame-Options: DENY",
    "path": "headers",
    "type": "config"
  }
]`;
